### ISSUE TITLE
performance layer normalization vs batch normalization

### ISSUE THREAD

**Issue by zamir1001** [oneapi-src/oneDNN#1058]
When I use layer normalization when the source memory descriptor uses the optimized aBcd8b format and for the stats memory descriptor I use the aBc8b format the execution time is very high compared to batch normalization.
Is this expected?

Thanks

**Comment by zamir1001** [comment]
Hi  @igorsafo ,

I was training a shufflenetV2 with the cifar10 dataset and replaced all the batch normalization layers with layer normalization.
The layer normalization was initialized with a fixed aBcd8b format and the performance was terrible.
I switched to a plain data format and also used a reorder as you pointed out. The performance was still lower than batchnorm but much better indeed.
Normally layer normalization is not used in this scenario I guess but there's a Funnel Relu activation (https://arxiv.org/abs/2007.11824)
where it was shown that using layer normalization instead of batch normalization showed a minor improvement in terms of accuracy but came with some additional computational cost. So they decided to just stick with batch normalization.
I was just experimenting with FRelu in combination with a shufflenetv2.

Thanks

Thanks

**Comment by zamir1001** [comment]
Hi, 
I read in the documentation you can fuse relu activation with layer normalization (like with batch normalization),
but when I try to create a forward primitive descriptor with the dnnl::normalization_flags::fuse_norm_relu flag I get an exception.

thanks

**Comment by dzarukin** [comment]
Hi @zamir1001, thanks for the question. Unfortunately, oneDNN doesn't. This is reflected on [documentation page](https://oneapi-src.github.io/oneDNN/dev_guide_layer_normalization.html) the way there's no information about such flag support or post-ops support. The only way as of now would be to create a separate eltwise primitive and apply it after layer normalization. Thanks.

**Comment by zamir1001** [comment]
Then there's some error in the documentation https://oneapi-src.github.io/oneDNN/group__dnnl__api__layer__normalization.html

Detailed Description:
...
The layer normalization primitives computations can be controlled by specifying different dnnl::normalization_flags values. For example, layer normalization forward propagation can be configured to either compute the mean and variance or take them as arguments. It can either perform scaling and shifting using gamma and beta parameters or not. Optionally, it can also perform a fused ReLU, which in case of training would also require a workspace.

thanks

**Comment by dzarukin** [comment]
Oh, this is definitely a miss from Doc side. We will fix it, thanks for reporting the issue.

**Comment by zamir1001** [comment]
No problem,
thanks for the clarification.

**Comment by zamir1001** [comment]
Hi 
Is there a possibility that in the future layer normalization could also have support for the optimized data format (nChw4,nChw8,nChw16) instead of only the plain as used in RNN today without having to use 2 reorders?

Thanks

**Comment by vpirogov** [comment]
In general oneDNN non-compute primitives should be optimized for the formats returned by compute primitives. So this is a performance optimizations gap.

**Comment by zamir1001** [comment]
Thanks for the information.
